vshard/util.lua

```diff
@@ -286,6 +286,37 @@ local function fiber_is_self_canceled()
     return not pcall(fiber.testcancel)
 end
 
+--
+-- Get min tuple from the index with the given key.
+--
+local index_min
+
+if version_is_at_least(2, 1, 0, nil, 0, 0) then
+
+-- At 1.10 index:min(key) works like 'get first tuple >= key', while at newer
+-- versions it works as 'get tuple having exactly the given key'. Need this
+-- workaround to simulate new behaviour.
+index_min = function(idx, key)
+    return idx:min(key)
+end
+
+else -- < 2.1.0
+
+local limit_1 = {limit = 1}
+index_min = function(idx, key)
+    local _, v = next(idx:select(key, limit_1))
+    return v
+end
+
+end
+
+--
+-- Check if the index has any tuple matching the key.
+--
+local function index_has(idx, key)
+    return index_min(idx, key) ~= nil
+end
```

Comment:
I guess it could be done with `index:count(key) ~= 0` (that is a bit GC-friendlier).